ground/src/main/java/com/google/android/ground/ui/datacollection/components/TaskViewFactory.kt

```diff
@@ -43,4 +44,21 @@ object TaskViewFactory {
     labelResId?.let { binding.headerLabel.setText(labelResId) }
     return TaskViewWithoutHeader(binding)
   }
+
+  /**
+   * creates a TaskView with a header that is an extension of the app bar
```

Comment:
nit: First letter of ktdoc should be capital and the sentence must end with a full stop.